Content.Server/Research/Disk/ResearchDiskSystem.cs

```diff
@@ -23,14 +23,22 @@ public override void Initialize()
         private void OnAfterInteract(EntityUid uid, ResearchDiskComponent component, AfterInteractEvent args)
         {
             if (!args.CanReach)
+            {
+                args.Handled = true;
                 return;
+            }
 
             if (!TryComp<ResearchServerComponent>(args.Target, out var server))
+            {
+                args.Handled = true;
                 return;
+            }
+
 
             _research.ModifyServerPoints(args.Target.Value, component.Points, server);
             _popupSystem.PopupEntity(Loc.GetString("research-disk-inserted", ("points", component.Points)), args.Target.Value, args.User);
             EntityManager.QueueDeleteEntity(uid);
+            args.Handled = true;
```

Comment:
1. You're marking it as handled when it's not actually handled.
2. Setting it in all codepaths is the same as just setting it at the start of the method anyway.

You should remove the first two.